### PR TITLE
Remove sys endpoint for enabling unseal recovery, switch to rekey

### PR DESCRIPTION
**Pull request by sgmiller**
To enable unseal recovery on a cluster that didn't have it, operators must
pass `-enable-unseal-recovery` to recovery rekey init.  Key share holders
must provide `-acknowledge-unseal-recovery` when providing shares.  Any
mismatch between the two aborts rekey.

If enabled, unseal recovery will work with the new keys after rekey.  If
disabled, any previous recovery unseal key material is deleted.

### CODE REVIEW

File: website/content/docs/concepts/seal.mdx [hunk 1]
```diff
@@ -272,6 +286,9 @@ to perform the migration. All unseal commands must specify the `-migrate` flag.
 Once the required threshold of recovery keys are entered, the recovery keys
 will be migrated to be used as unseal keys.
 
+Note that you may also migrate from an Auto Unseal in 
+[unseal recovery mode](#unseal-recovery-mode)
```
Comment (cipherboy) [inline website/content/docs/concepts/seal.mdx:290]:
```suggestion
[unseal recovery mode](#unseal-recovery-mode).
```

File: vault/core.go [hunk 2]
```diff
@@ -4051,3 +4051,30 @@ func (c *Core) GetRaftAutopilotState(ctx context.Context) (*raft.AutopilotState,
...
+	})
+	recoverySeal.SetCore(c)
+	if err := recoverySeal.SetStoredKeys(ctx, [][]byte{barrierKey}); err != nil {
+		return fmt.Errorf("failed to store recover unseal keys: %v", err)
+	}
+	return nil
+}
+
+func (c *Core) UnsealRecoveryEnabled(ctx context.Context) (bool, error) {
+	// TODO: Return false on ENT
```
Comment (cipherboy) [inline vault/core.go:4071]: Could you just add a `constants.IsEnterprise || (...)` here?

**Review by cipherboy — COMMENTED** [review]
I _think_ this makes sense to me, but it'd be good to get someone else's eyes on it.

File: vault/core.go (hunk 2 above)
Comment (sgmiller) [reply vault/core.go:4071]: Oh sweet, TiL.

**Pull request closed without merging**